font/src/ft/mod.rs

```diff
@@ -253,31 +246,16 @@ impl FreeTypeRasterizer {
         let primary_font = pattern.render_prepare(config, primary_font);
 
         // Hash pattern together with request pattern to include requested font size in the hash
-        let primary_font_id = FontID::new(hash, primary_font.hash());
-
-        // Reload already loaded faces and drop their fallback faces
-        let font_key = if let Some(font_key) = self.keys.remove(&primary_font_id) {
-            let fallback_list = self.fallback_lists.remove(&font_key).unwrap_or_default();
-
-            for fallback_font in &fallback_list.list {
-                if let Some(ff_key) = self.keys.get(&fallback_font.id) {
-                    // Skip primary fonts, since these are all reloaded later
-                    if !self.fallback_lists.contains_key(&ff_key) {
-                        self.faces.remove(ff_key);
-                        self.keys.remove(&fallback_font.id);
-                    }
-                }
-            }
+        let primary_font_key = FontKey::from_pattern_hashes(hash, primary_font.hash());
 
-            let _ = self.faces.remove(&font_key);
-            Some(font_key)
-        } else {
-            None
-        };
+        // Return font if we already loaded it
+        if self.faces.get(&primary_font_key).is_some() {
+            return Ok(primary_font_key);
+        }
```

Comment:
Thinking about this early return, there might be an issue with fallback faces, right?

In `self.faces` both primary and fallback faces are stored. However, a primary face requires the additional computation we do in this method to load the fallback list.

So it seems like what we should do is check if the font is already loaded **and** if a fallback list exists for this font.

However even if the fallback list does not exist, we should be able to still skip the call to `face_from_pattern`. Just not the entire rest of this function.

Reply:
>Thinking about this early return, there might be an issue with fallback faces, right?

Yeah, exactly, forgot about this case.